src/reader.js

```diff
@@ -148,8 +148,13 @@ class Reader extends EventEmitter {
 
     // Trigger a query of the configured `lookupdHTTPAddresses`.
     const endpoint = this.roundrobinLookupd.next();
-    lookup(endpoint, this.topic, (err, nodes = []) =>
-      nodes.map(n => this.connectToNSQD(n.broadcast_address, n.tcp_port)));
+    console.log(endpoint);
+    lookup(endpoint, this.topic, (err, nodes = []) => {
+      if(err){
```

Comment:
The possibility you are thinking about is the following connection params:
```
{
nsqdTCPAddresses:['localhost'],
lookupdHTTPAddresses:['someOtherServer']
}
```
In this circumstances nsqjs would connect to the nsqd on the same host and throw an error when it can't connect to the lookupd. As this method is only called on startup (with a delay) I still think it is not too bad to throw an error.


If you belive it isn't ok an other option would be:
it should be checked whether the provided lookupd address is in fact a lookup-instance by checking either `/ping` or `/info` and only output a console error instead of an "real" error.